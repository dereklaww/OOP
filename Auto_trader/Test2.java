import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class Test2 {

    public static void main(String[] args) {
        Market market = new Market();

        Trader t1 = new Trader("ABC", 10.00);

        Trader t2 = new Trader("DEF", 10.00);
        t2.importProduct("nike", 100.00);
        System.out.println(t2.getAmountStored​("nike"));


        Order buy_o1 = new Order("nike",true,5.00, 10.00, t1, "nike_1");
        Order buy_o3 = new Order("nike",true,10.00, 8.00, t1, "nike_2");
        Order buy_o5 = new Order("nike",true,15.00, 5.00, t1, "nike_3");
        Order buy_o6 = new Order("nike",true,25.00, 10.00, t1, "nike_4");
        Order buy_o2 = new Order("adidas",true,10.00, 10.00, t1, "adidas_1");
        Order buy_o4 = new Order("adidas",true,5.00, 5.00, t1, "adidas_2");

    
        market.buyBook.add(buy_o1);
        market.buyBook.add(buy_o2);
        market.buyBook.add(buy_o3);
        market.buyBook.add(buy_o4);
        market.buyBook.add(buy_o5);
        market.buyBook.add(buy_o6);


        Order sellOrder = new Order("nike",false,35.00, 2.00, t2, "nike_sell");

        List<Trade> completed_trades =market.placeSellOrder​(sellOrder);

        // System.out.println(completed_trades);
    }
    
}
