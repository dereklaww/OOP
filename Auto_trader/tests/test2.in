add trader1 100
add trader2 100
import trader1 Nike 100
import trader1 Adidas 200
sell trader2 Nike 80 10
sell trader1 Nike 80 10
amount trader1 Nike
buy trader2 Adidas 50 5
order 0000
order 0001
order 0002
sell trader1 Adidas 20 10
sell trader1 Adidas 30 5
sell trader1 Adidas 50 5
sell trader3 Adidas 10 10
book sell
book buy
buy trader2 Nike 50 10
buy trader2 Nike 100 10
buy trader3 Nike 100 10
book buy
book sell
exit