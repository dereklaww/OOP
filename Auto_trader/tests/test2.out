$ Success.
$ Success.
$ Trader now has 100.00 units of Nike.
$ Trader now has 200.00 units of Adidas.
$ Order could not be placed onto the market.
$ No trades could be made, order added to sell book.
$ 20.00
$ No trades could be made, order added to buy book.
$ Order is not present in either order book.
$ 0001: SELL 80.00xNike @ $10.00
$ 0002: BUY 50.00xAdidas @ $5.00
$ No trades could be made, order added to sell book.
$ Product sold in entirety, trades as follows:
trader1->trader2: 30.00xAdidas for $5.00.
$ Product sold in part, trades as follows:
trader1->trader2: 20.00xAdidas for $5.00.
$ No such trader in the market.
$ 0001: SELL 80.00xNike @ $10.00
0003: SELL 20.00xAdidas @ $10.00
0005: SELL 30.00xAdidas @ $5.00
$ The buy book is empty.
$ Product bought in entirety, trades as follows:
trader1->trader2: 50.00xNike for $10.00.
$ Product bought in part, trades as follows:
trader1->trader2: 30.00xNike for $10.00.
$ No such trader in the market.
$ 0007: BUY 70.00xNike @ $10.00
$ 0003: SELL 20.00xAdidas @ $10.00
0005: SELL 30.00xAdidas @ $5.00
$ Have a nice day.